lib/spack/spack/test/cmd/ci.py

```diff
@@ -1456,3 +1550,122 @@ def test_ci_generate_read_broken_specs_url(tmpdir, mutable_mock_env_path,
+
+            root_spec = None
+            job_spec = None
+
+            for h, s in env.specs_by_hash.items():
+                print('{0} -> {1}'.format(s.name, h))
+                if s.name == 'archive-files':
+                    root_spec = s
+                    job_spec = s
+
+            job_spec_yaml_path = os.path.join(
+                working_dir.strpath, 'archivefiles.yaml')
+            with open(job_spec_yaml_path, 'w') as fd:
+                fd.write(job_spec.to_yaml(hash=ht.full_hash))
+
+            root_spec_yaml_path = os.path.join(
+                working_dir.strpath, 'root.yaml')
+            with open(root_spec_yaml_path, 'w') as fd:
+                fd.write(root_spec.to_yaml(hash=ht.full_hash))
+
+            artifacts_root = os.path.join(working_dir.strpath, 'scratch_dir')
+            pipeline_path = os.path.join(artifacts_root, 'pipeline.yml')
+
+            ci_cmd('generate', '--output-file', pipeline_path,
+                   '--artifacts-root', artifacts_root)
+
+            job_name = ci.get_job_name(
+                'specs', False, job_spec, 'test-debian6-core2', None)
+
+            print('job name: {0}'.format(job_name))
+
+            repro_file = os.path.join(working_dir.strpath, 'repro.json')
+            repro_details = {
+                'job_name': job_name,
+                'job_spec_yaml': 'archivefiles.yaml',
+                'root_spec_yaml': 'root.yaml'
+            }
+            with open(repro_file, 'w') as fd:
+                fd.write(json.dumps(repro_details))
+
+            install_script = os.path.join(working_dir.strpath, 'install.sh')
+            with open(install_script, 'w') as fd:
+                fd.write('#!/bin/bash\n\n#fake install\nspack install blah\n')
+
+            git = exe.which("git", required=True)
+            spack_git_path = spack.paths.prefix
+            with fs.working_dir(spack_git_path):
+                git_log_out = git('log', '-n', '2', output=str, error=os.devnull)
+
+            regex = re.compile(r"^commit\s([^\s]+$)", re.MULTILINE)
+            m = regex.findall(git_log_out)
+
+            spack_info_file = os.path.join(
+                working_dir.strpath, 'spack_info.txt')
+            with open(spack_info_file, 'w') as fd:
+                fd.write('\nMerge {0} into {1}\n\n'.format(m[1], m[0]))
+
+    def fake_download_and_extract_artifacts(url, work_dir):
+        print('fake_download_and_extract_artifacts')
+
+    monkeypatch.setattr(ci, 'download_and_extract_artifacts',
+                        fake_download_and_extract_artifacts)
+
+    rep_out = ci_cmd('reproduce-build',
+                     'https://some.domain/api/v1/projects/1/jobs/2/artifacts',
+                     '--working-dir',
+                     working_dir.strpath,
+                     output=str)
+
+    expect_out = 'docker run --rm -v {0}:{0} -ti {1}'.format(
+        working_dir.strpath, image_name)
+
+    print('spack ci reproduce-build ...')
```

Comment:
It looks like there are still some debug print statements in this test

Reply:
In general I don't try to remove those since they don't get printed unless there's a failure.  If it's important to remove this, then should I also go through and remove all the other print statements from this module?

Reply:
I guess it doesn't matter -- I've always made sure to remove them since it's not part  of the test proper.